src/pages/Navgurukul/merakiAdmission/index.js

```diff
@@ -0,0 +1,211 @@
+    mobileNumber: "",
+  });
+
+  const [mobile, setMobile] = useState("");
+  const [enrolmentKey, setEnrolmentKey] = useState("");
+
+  const [partnerId, setPartnerId] = useState("");
+
+  let userToken = localStorage.getItem("Token");
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/users/me`,
+      headers: {
+        accept: "application/json",
+        Authorization: userToken,
+      },
+    }).then((res) => {
+      setPartnerId(res.data.user.partner_id);
+    });
+  }, []);
+
+  const testUrl = "http://dev-join.navgurukul.org/k/";
+
+  const generateTestLink = async () => {
+    try {
+      const mobile = `0${userDetails.mobileNumber}`;
+      const dataURL = `${process.env.REACT_APP_CHANAKYA_BASE_URL}helpline/register_exotel_call`;
+      const response = await axios.get(dataURL, {
+        params: {
+          ngCallType: "getEnrolmentKey",
+          From: mobile,
+          partner_id: partnerId ? partnerId : null,
+        },
+      });
+      setEnrolmentKey(response.data.key);
+      const params = {
+        firstName: userDetails.firstName,
+        middleName: userDetails.middleName,
+        lastName: userDetails.lastName,
+        mobileNumber: userDetails.mobileNumber,
+      };
+      const queryString = Object.keys(params)
+        .map((filter) => {
+          if (params[filter]) {
+            return `${filter}=${params[filter]}`;
+          }
+          return null;
+        })
+        .filter((item) => item)
+        .join("&");
+
+      const url = `${testUrl}${enrolmentKey}?${queryString}`;
+
+      window.open(url, "_blank");
+      return response;
+    } catch (e) {}
+  };
+
+  const giveTest = async () => {
+    if (
+      !userDetails.firstName ||
+      !userDetails.lastName ||
+      !userDetails.mobileNumber
+    ) {
+      toast.error(
+        "To attempt the test, it is compulsory to enter your First Name, Last Name and Mobile Number. Middle Name is optional, you can choose not to enter.",
+        {
+          position: toast.POSITION.BOTTOM_RIGHT,
+        }
+      );
+      return;
+    }
+    if (userDetails.mobileNumber.toString().length !== 10) {
+      toast.error("Please give 10 digits of the mobile number.", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+      });
+      return;
+    }
+    await isDuplicate();
+  };
+
+  const isDuplicate = () => {
+    const { firstName, middleName, lastName, mobileNumber } = userDetails;
+    axios
+      .get(`${process.env.REACT_APP_CHANAKYA_BASE_URL}/check_duplicate`, {
+        params: {
+          Name: firstName.concat(" ", middleName, lastName),
+          Number: mobileNumber,
+        },
+      })
+      .then(async (data) => {
+        const response = data.data.data;
+        const stage = response.pendingInterviewStage;
+        const url = `https://admissions.navgurukul.org/check_duplicate/Name=${firstName}${middleName}${lastName}&Number=${mobileNumber}&Stage=${stage}`;
```

Comment:
It seems preferable to use `URLSearchParams` here to explicitly handle encoding of reserved characters (e.g., if a last name contains an apostrophe).  So you could do the following:
```suggestion
        const ADMISSIONS_URL = `https://admissions.navgurukul.org`;
        const url = `${ADMISSIONS_URL}/check_duplicate/` +
            new URLSearchParams({
              Name: `${firstName}${middleName}${lastName}`, 
              Number: mobileNumber, 
              Stage: stage
            }).toString();
```
But you'd remove `ADMISSIONS_URL` in place of an import statement if you move it to a config file.